os/kernel/fault_manager/fault_manager.c

```diff
@@ -0,0 +1,285 @@
+ *   This function prints applciation pid and its corresponding application path
+ *   which are being registered for fault manager.
+ *
+ * Input parameters:
+ *   None
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+void fm_printapptable(void)
+{
+	int i;
+
+	lldbg("-------------------------------------\n");
+	lldbg(" APP_ID	APP_PATH\n");
+	for (i = 0; i < CONFIG_NUM_APPS; i++) {
+		lldbg(" %d\t%s\n", reg_table[i].appid, reg_table[i].app_path);
+	}
+	lldbg("-------------------------------------\n");
+
+}
+
+/****************************************************************************
+ * Name: get_appid
+ *
+ * Description:
+ *   This function will traverse the task/thread tree and return the root app pid.
+ *   If fault happened in any of the thread/task, it will call this function to get
+ *   root application pid for the same.
+ *
+ * Input parameters:
+ *   pid   -   The pid of the task/thread, whose root app pid is required
+ *
+ * Returned Value:
+ *   pid   -   pid of the root application of input pid
+ *
+ ****************************************************************************/
+pid_t get_appid(pid_t pid)
+{
+	/* TODO: Implementation of traverser of the tree to get root app pid */
+	return pid;
+}
+
+/****************************************************************************
+ * Name: kill_app
+ *
+ * Description:
+ *   This function will kill all the task/thread created by the root application
+ *   i.e input pid.
+ *   If any fault occurs in the system, fault manager will call this function to
+ *   kill all the thread/task created by the input application.
+ *
+ * Input parameters:
+ *   pid   -   The pid of the applciation, whoes all childs to be killed
+ *
+ * Returned Value:
+ *   Zero (OK) on success; otherwise -1 (ERROR) value is returned.
+ *
+ ****************************************************************************/
+int kill_app(pid_t pid)
+{
+	int ret;
+
+	/* TODO: Implementation of traverser of tree and kill all the task/thread
+	 * of the root application
+	 */
+	ret = kill(pid, SIGWORK);
+	if (ret == -1) {
+		return ERROR;
+	}
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: fault_manager_init
+ *
+ * Description:
+ *   This function will receive the faulty application's pid and check if it is one
+ *   of the registered application for fault manager.
+ *   If the application is registered then it will be restarted after killing its
+ *   child processes, and if it is not registered then board will be rebooted.
+ *
+ * Input parameters:
+ *   None
+ *
+ * Returned Value:
+ *   Zero (OK) on success; otherwise -1 (ERROR) value is returned.
+ *
+ * Assumptions:
+ *    All the applications of the system will be registered for the fault manager
+ *
+ ****************************************************************************/
+int fault_manager_init(int argc, char *argv[])
```

Comment:
Usually `init` means initialization. Is this approprite to call init? This looks main function of fault manager. This function should be running always. It does not quit after init.

Reply:
I have modified the name from fault_manager_init() to fault_manager()